contracts/polyx-vesting/lib.rs

```diff
@@ -0,0 +1,168 @@
+        #[ink(message)]
+        pub fn duration(&self) -> u128 {
+            self.duration
+        }
+
+        /// Returns the start timestamp.
+        #[ink(message)]
+        pub fn start(&self) -> u128 {
+            self.start
+        }
+
+        /// Returns the beneficiary address.
+        #[ink(message)]
+        pub fn beneficiary(&self) -> AccountId {
+            self.beneficiary
+        }
+
+        /// Returns the amount of POLYX already released.
+        #[ink(message)]
+        pub fn released(&self) -> Balance {
+            self.released
+        }
+
+        /// Returns the amount of releasable POLYX.
+        #[ink(message)]
+        pub fn releasable(&self) -> Balance {
+            self.vested_amount(self.env().block_timestamp().into())
+                .saturating_sub(self.released())
+        }
+
+        #[ink(message)]
+        pub fn get_time(&self) -> Balance {
+            self.env().block_timestamp().into()
+        }
+
+        #[ink(message)]
+        pub fn get_balance(&self) -> Balance {
+            self.env().balance()
+        }
+
+        /// Release the native token (POLYX) that have already vested.
+        #[ink(message)]
+        pub fn release(&mut self) -> Result<()> {
+            let amount = self.releasable();
+            assert!(amount > 0, "insufficient funds!");
+            self.released += amount;
+            Self::env().emit_event(PolyxReleased { value: amount });
+            if self.env().transfer(self.env().caller(), amount).is_err() {
+                Err(Error::InsufficientBalance)
+            } else {
+                Ok(())
+            }
+        }
+
+        /// Calculates the amount of tokens that has already vested.
+        #[ink(message)]
+        pub fn vested_amount(&self, timestamp: u128) -> Balance {
+            self.vesting_schedule(self.env().balance() + self.released, timestamp)
+        }
+
+        /// This returns the amount vested.
+        fn vesting_schedule(&self, total_allocation: u128, timestamp: u128) -> Balance {
+            if timestamp < self.start {
+                return 0;
+            } else if timestamp > self.start + self.duration {
+                return total_allocation;
+            } else {
+                return (total_allocation * (timestamp - self.start)) / self.duration;
+            }
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        /// Imports all the definitions from the outer scope so we can use them here.
+        use super::*;
+
+        /// Imports `ink_lang` so we can use `#[ink::test]`.
+        use ink_lang as ink;
+
+        /// We test if the new constructor does its job.
+        #[ink::test]
+        fn new_works() {
+            // Constructor works.
+            let polyx_vesting = PolyxVesting::new(AccountId::from([0x01; 32]), 5, 20);
+            // Ensure the values are stored correctly
+            assert_eq!(polyx_vesting.beneficiary(), AccountId::from([0x01; 32]));
```

Comment:
Please use the default accounts (Alice, Bob, Charlie, etc...) instead of raw bytes.  You can get them from: https://docs.rs/ink_env/latest/ink_env/test/fn.default_accounts.html